rc/lsp.kak

```diff
@@ -1983,6 +2036,15 @@ define-command -hidden lsp-mappings-enable -params 1 -docstring "Add LSP mapping
 }
```

Comment:
So I dropped this for now, looking to find a more general solution eventually.

I also found a problem with overloading `<c-n>` this way, so I've mapped `<tab>` as well, which seems nicer to use.
If you like `<c-n>` I'll keep it but otherwise I'd stick to `<tab>` only.

Reply:
Mapping tab breaks my tab indent, and i assume it can break a lot of things for other people as well, so unless we find a way to make it completely transparent, i don't think it is a good idea as a default. `<c-n>` is easier, as i assume it is less likely to be in use for other things already.

Reply:
> Mapping tab breaks my tab indent

it indents just fine for me if there is no placeholder.
I assume you're seeing breakage because you created a mapping for tab?

`<c-n>` is a problematic default because we can't use it while the completion pager is active.

I think, whatever hotkey we end up using, we should tell users
to do something like this if they'd normally remap the key:

```kak
map global insert <tab> '<a-;>: tab-in-insert-mode<ret>'
define-command -hidden tab-in-insert-mode %{
    try %{
        lsp-snippets-select-next-placeholders
    } catch %{
        # do whatever tab normally does, the default is:
        execute-keys -with-hooks %arg{1}
    }
}
# similar for "normal" instead of "insert"
```

Reply:
I've merged all commits except for the one that adds the mappings. I'll try both mappings for a bit